sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -2007,4 +2008,125 @@ class RegexUnsupportedException(message: String, index: Option[Int])
       case _ => message
     }
   }
+}
+
+sealed trait RegexOptimizationType
+object RegexOptimizationType {
+  case class StartsWith(literal: String) extends RegexOptimizationType
+  case class Contains(literal: String) extends RegexOptimizationType
+  case class PrefixRange(literal: String, charsRange: (Int, Int), length: Int) 
+    extends RegexOptimizationType
+  case object NoOptimization extends RegexOptimizationType
+}
+
+object RegexRewriteUtils {
+
+  private def removeBrackets(astLs: collection.Seq[RegexAST]): collection.Seq[RegexAST] = {
+    astLs match {
+      case collection.Seq(RegexGroup(_, term, None)) => removeBrackets(term.children())
+      case _ => astLs
+    }
+  }
+  private def getPrefixRangePattern(astLs: collection.Seq[RegexAST]): 
+      Option[(String, Int, Int, Int)] = {
+    val endsWithRange = astLs.last match {
+      case RegexRepetition(
+          RegexCharacterClass(false,ListBuffer(RegexCharacterRange(a,b))), 
+          quantifier) => {
+        val (start, end) = (a, b) match {
+          case (RegexChar(start), RegexChar(end)) => (start, end)
+          case _ => return None
+        }
+        val x = quantifier match {
+          case QuantifierVariableLength(start, _) => start
```

Comment:
I don't think this is correct.  We are ignoring the end and only using start.

Reply:
ditto

Reply:
This has still not been addressed. Also we are reusing the variable start for both a length and a character. Could we please update the names so it is clear which is which?

Reply:
Updated the names and added some comments. If a string contains `literal[a-b]{x,y}` it must contains `literal[a-b]{x}` first so `y` here is useless.